pkg/recipes/driver/terraform.go

```diff
@@ -135,13 +135,28 @@ func (d *terraformDriver) Delete(ctx context.Context, opts DeleteOptions) error
 			logger.Info(fmt.Sprintf("Failed to cleanup Terraform execution directory %q. Err: %s", requestDirPath, err.Error()))
 		}
 	}()
-
+	// Add credential information to .gitconfig if module source is of type git.
+	if strings.HasPrefix(opts.Definition.TemplatePath, "git::") && !reflect.DeepEqual(opts.BaseOptions.Secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
```

Comment:
Should this be a function? Like `IsOfTypeGit` or something like that?

Reply:
we are making it a function in a different PR which will go on top of this one, which add supports for nested modules. because some changes are going to be done to the add/unset git config functions.

Reply:
Could you also please post the link to the PR here?